expertise/create_dataset.py

```diff
@@ -85,13 +89,50 @@ def deduplicate_publications(self, publications):
                 deduplicated.append(pub)
 
         return deduplicated
+    
+    def get_pub_weight(self, pub=None, weight_specification=[]):
+        def _matches(venue_spec, venueid, domain):
+            in_openreview = venueid == domain and domain in self.venue_list
+            ## Papers allowed: accepted papers from an OpenReview venue
+            ## not in_openreview: DBLP papers (venueid =/= domain) and non-accepted papers (domain not in venue_list)
+
+            return (
+                ('prefix' in venue_spec and in_openreview and venueid.startswith(venue_spec['prefix'])) or
+                ('value' in venue_spec and in_openreview and venueid == venue_spec['value']) or
+                ('inOpenReview' in venue_spec and in_openreview and venue_spec['inOpenReview']) or
+                ('inOpenReview' in venue_spec and not in_openreview and not venue_spec['inOpenReview'])
+            )
+        
+        venueid = pub.content.get('venueid', '')
+        if isinstance(venueid, dict):
+            venueid = venueid.get('value')
+        
+        ## If weight specification but no venueid, return default weight
+        if not venueid:
+            return 1
+
+        # Get domain from either domain field or invitation prefix
+        domain = getattr(pub, 'domain', None)
+        if domain is None:
+            domain = pub.invitation.split('/-/')[0]
+            
+        # Find matching weight specification
+        current_weight, current_order = 1, None ## Default weight one
+        for idx, venue_spec in enumerate(weight_specification):
+            if _matches(venue_spec, venueid, domain):
+                order = venue_spec.get('order', idx) ## Support optional order, fallback to index
+                if current_order is None or order <= current_order:
+                    current_weight = venue_spec['weight']
+                    current_order = order
```

Comment:
What is `order` used for?

Reply:
I saw above your comment :) 
If you can share an example, it would be helpful

Reply:
Implementing value > prefix > articleSubmittedToOpenReview rule